tests/Unit/NumericalAlgorithms/Interpolation/Test_PredictedZeroCrossing.cpp

```diff
@@ -47,21 +48,38 @@ void test_predicted_zero_crossing_datavector() {
     y_values.push_back(tmp);
   }
 
-  DataVector compute_zero_crossing_value =
+  const auto compute_zero_crossing_value =
       intrp::predicted_zero_crossing_value(x_values, y_values);
 
   Approx custom_approx = Approx::custom().epsilon(1e-6);
-  CHECK_ITERABLE_CUSTOM_APPROX(compute_zero_crossing_value,
+  CHECK_ITERABLE_CUSTOM_APPROX(compute_zero_crossing_value.best_fit,
                                expected_zero_crossing_value, custom_approx);
 }
 
 void test_predicted_zero_crossing_indeterminate() {
   // Here we set up points so that the error bars are large enough
-  // that it is not clear whether the zero crossing is in the
-  // past or in the future.
+  // that it is not clear whether the zero crossing is in the past or
+  // in the future.  So the the two returned bounds should be of
+  // opposite signs.
   std::vector<double> x_values{0.0, 1.0, 2.0, 3.0};
   std::vector<double> y_values{1.0, 2.1, 1.0, 2.0};
-  CHECK(intrp::predicted_zero_crossing_value(x_values, y_values) == 0.0);
+  const auto zero_crossing =
+      intrp::predicted_zero_crossing_value(x_values, y_values);
```

Comment:
```
  zero_crossing.lower_bound := -3.65971046088218799
  zero_crossing.best_fit := -6.52631578947368318
  zero_crossing.upper_bound := 15.80900282881401964
```
You can perhaps argue that this result isn't wrong, but the meaning is certainly unintuitive.  (Look at the ordering.)  At minimum you need some explanation in the documentation.  This may not be a better return type than the old one.

Reply:
Oh, wow. I didn't expect this either.  And I think the bounds are actually wrong.

But I think I know what is going on: currently I return the minimum of (x0,x1,x2,x3) as lower_bound and the maximum of (x0,x1,x2,x3) as upper_bound.  But the error bars are so large that e.g. (slope+delta_slope) has a different sign than (slope-delta_slope).   So let's consider x0,x1,x2, and x3 to be functions of delta_slope and delta_intercept, so that x0=x1=x2=x3=x_best_fit when delta_slope=delta_intercept=0.  Then as delta_slope varies from zero to its actual value, at least two of the functions x0,x1,x2,x3 will pass through +/- infinity and switch sign.  When this happens, either lower_bound or upper_bound should really be +/- infinity instead of what I return now.

Possible solutions:

1. Revert back to the previous return type and the previous logic (which is all that is needed for the only current use case).
1. Work out the logic to correctly compute lower_bound and upper_bound, and in some cases return lower_bound=-infinity and/or upper_bound=+infinity.   Note that the code that calls PredictedZeroCrossing (which now checks that lower_bound*upper_bound>0 as a test that a zero-crossing is well determined) would also need to check if the bounds are finite in this test.

Thoughts?

Reply:
Okay this seems to be not worth the effort currently.  So reverting to the previous return type is fine, but please add documentation (and an ASSERT?) that the input x_values are supposed to be negative times before the current time zero (i.e. basically the comment you wrote above two days ago) so no one tries to use this function to do something it can't handle correctly